File: app/schemas/market.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class StockDailyBase(BaseModel):
    ts_code: str
    trade_date: date
    open: float | None = None
    high: float | None = None
    low: float | None = None
    close: float | None = None
    pre_close: float | None = None
    change: float | None = None
    pct_chg: float | None = None
    vol: float | None = None
    amount: float | None = None


class StockDailyCreate(StockDailyBase):
    pass


class StockDailyInDB(StockDailyBase):
    id: int
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class StockDailyOut(StockDailyBase):
    id: int
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class IndexDailyBase(BaseModel):
    ts_code: str
    trade_date: date
    open: float | None = None
    high: float | None = None
    low: float | None = None
    close: float | None = None
    pre_close: float | None = None
    change: float | None = None
    pct_chg: float | None = None
    vol: float | None = None
    amount: float | None = None


class IndexDailyCreate(IndexDailyBase):
    pass


class IndexDailyInDB(IndexDailyBase):
    id: int
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class IndexDailyOut(IndexDailyBase):
    id: int
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/market.ts
import client from './client';

export interface StockDaily {
  id: number;
  ts_code: string;
  trade_date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  pre_close: number;
  change: number;
  pct_chg: number;
  vol: number;
  amount: number;
  created_at: string;
  updated_at: string;
}

export interface IndexDaily {
  id: number;
  index_code: string;
  trade_date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  pre_close: number;
  change: number;
  pct_chg: number;
  vol: number;
  amount: number;
  created_at: string;
  updated_at: string;
}

export const marketApi = {
  getStockDaily: (tsCode: string, startDate: string, endDate: string) =>
    client.get<StockDaily[]>('/market/stock-daily', { params: { ts_code: tsCode, start_date: startDate, end_date: endDate } }),
  getIndexDaily: (indexCode: string, startDate: string, endDate: string) =>
    client.get<IndexDaily[]>('/market/index-daily', { params: { index_code: indexCode, start_date: startDate, end_date: endDate } }),
};
